spec/test/stored_queries/test_query.py

```diff
@@ -0,0 +1,436 @@
+    CLIENT = ArangoClient(hosts=CONFIG["host"])
+    DB = CLIENT.db("ci", username=CONFIG["username"], password=CONFIG["password"])
+except Exception as e:
+    help = """
+Please set host URL, username, and password in arango_live_server_config.json, e.g.,
+{
+    "username": "doe_j",
+    "password": "cat-sat-hat",
+    "host": "http://10.58.1.211:8532"
+}
+Note: if you are on a local machine
+you may have to proxy into the live ArangoDB server first, e.g.,
+`ssh -L 8532:10.58.1.211:8532 j_doe@login1.berkeley.kbase.us`
+Then, the url would be `http://localhost:8532`
+"""
+    print(help)
+    raise (e)
+NCBI_TAXON = DB.collection("ncbi_taxon")
+
+# Load the queries
+QUERY = json_validation.load_json_yaml(STORED_QUERY_FP)["query"]
+QUERY_NO_SORT = json_validation.load_json_yaml(STORED_QUERY_NO_SORT_FP)["query"]
+
+LIMIT = 20
+NOW = time.time() * 1000
+
+# Load/cache the scinames
+if os.path.isfile(SCINAMES_LATEST_FP):
+    with open(SCINAMES_LATEST_FP) as fh:
+        SCINAMES_LATEST = json.load(fh)
+else:
+    print("Fetching latest NCBI scinames ...")
+    try:
+        taxa_all = list(NCBI_TAXON.all())
+    except ReadTimeout:
+        print("Sorry, there is a read timeout. Please try again on a different machine")
+        sys.exit()
+    SCINAMES_LATEST = [
+        taxa["scientific_name"]
+        for taxa in taxa_all
+        if (taxa["rank"] in ["species", "strain"] or taxa["strain"])
+        and taxa["created"] <= NOW
+        and NOW <= taxa["expired"]
+    ]
+    with open(SCINAMES_LATEST_FP, "w") as fh:
+        json.dump(SCINAMES_LATEST, fh)
+
+
+def use_sort(search_text):
+    return len(search_text) > 3
+
+
+def is_simple(search_text):
+    """
+    Somewhat arbitrary determination of whether a fulltext's search text is "simple"
+    relative to its search time
+    """
+    return len(search_text.split()) == 2 and all(
+        [tok.isalnum() and len(tok) >= 3 for tok in search_text.split()]
+    )
+
+
+def jprint(jo, dry=False):
+    txt = json.dumps(jo, indent=3)
+    if dry:
+        return txt
+    else:
+        print(txt)
+
+
+def fulltext_search_ncbi_scinames(search_text):
+    """"""
+    cursor = DB.aql.execute(
+        QUERY if use_sort(search_text) else QUERY_NO_SORT,
+        bind_vars={
+            "search_text": search_text,
+            "ts": NOW,
+            "offset": None,
+            "limit": LIMIT,
+            "select": ["scientific_name"],
+        },
+    )
+    return {
+        "results": [e["scientific_name"] for e in list(cursor.batch())],
+        **cursor.statistics(),
+    }
+
+
+def get_search_text_samplings(
+    resample=True,
+    cap_scinames=2000,
+    cap_scinames_prefixes=5000,
+):
+    """
+    Get samplings of scinames or prefixes thereof to gauge execution time
+
+    Things to include:
+    * Simple genus/species epithets with two non-short words
+    * "Wild" scientific names, defined as the exclusion of the simple scientific names
```

Comment:
I think that the "simple scientific names" are usually called "binomial" identifiers, so you could call the wild ones "non-binomial" or something like "complex" or "non-canonical"

Reply:
yeah ... didn't think this would end up in the repo hehe